### DISCUSSION TITLE
UV Mapping on a specific area

### DISCUSSION THREAD

**Discussion by DanieleMagri** [google/model-viewer#3514] [Q&A]
I followed[ this example](https://modelviewer.dev/examples/scenegraph/#swapTextures), it changes the textures for me, but on the whole model, is there a way to change the textures on an area of ​​the model based on UV mapping? Otherwise I have to use Three.js to customize the model and then export the GLB model in model viewer every time I change texture, I would like to avoid having to do this, although it works well as a solution, because the texture change would be faster if done using the model viewer API.

Should i redo all textures considering the whole model? At the moment the textures refer to that specific area, so the UV Mapping was created based only on that sub-area, not on the whole model, I hope I have explained myself well.

**Comment by elalish** [comment]
The granularity of our API is per glTF material. If you want to change the texture on a portion of the model, then that portion should have its own material.

**Reply by DanieleMagri** [reply]
So to be clearer, would it be possible to create something [like this](https://tympanus.net/Tutorials/3DModelCustomizer/) where you can change the texture by selecting a part of the mesh?

**Reply by elalish** [reply]
Yes, see this [example](https://modelviewer.dev/examples/scenegraph/#pickMaterialExample), for instance. We're just changing color factors, but textures would work the same way.
